tracing/tracing-test.sh

```diff
@@ -72,112 +112,133 @@ stop_jaeger()
 	docker rm -f "${jaeger_docker_container_name}"
 }
 
-# Returns status from Jaeger web UI
-get_jaeger_status()
+get_jaeger_traces()
 {
 	local service="$1"
-
 	[ -z "$service" ] && die "need jaeger service name"
 
-	local attempt=0
-	local status=""
+	local traces_url="http://${jaeger_server}:${jaeger_ui_port}/api/traces?service=${service}"
+	curl -s "${traces_url}" 2>/dev/null
+}
 
-	while [ $attempt -lt 10 ]
-	do
-		status=$(curl -s "http://${jaeger_server}:${jaeger_ui_port}/api/traces?service=${service}" 2>/dev/null)
-		local ret=$?
+get_trace_summary()
+{
+	local status="$1"
+	[ -z "$status" ] && die "need jaeger status JSON"
 
-		[ "$ret" -eq 0 ] && [ -n "$status" ] && break
+	echo "${status}" | jq -S '.data[].spans[] | [.spanID, .operationName] | @sh'
+}
 
-		attempt=$((attempt+1))
-		sleep 1
-	done
+get_span_count()
+{
+	local status="$1"
+	[ -z "$status" ] && die "need jaeger status JSON"
 
-	echo "$status"
+	# This could be simplified but creating a variable holding the
+	# summary is useful in debug mode as the summary is displayed.
+	local trace_summary=$(get_trace_summary "$status" || true)
+
+	local count=$(echo "${trace_summary}" | wc -l)
+
+	[ -z "$count" ] && count=0
+
+	echo "$count"
 }
```

Comment:
If `trace_summary` is empty (or `true`), `wc -l` will return 1 and `count` will never be 0. Should this be `wc -w`?